src/System.Net.WebSockets.Client/tests/SendReceiveTest.cs

```diff
@@ -157,45 +157,42 @@ public class SendReceiveTest : ClientWebSocketTestBase
                 var recvBuffer = new byte[100];
                 var recvSegment = new ArraySegment<byte>(recvBuffer);
 
-                try
+                for (int i = 0; i < tasks.Length; i++)
                 {
-                    for (int i = 0; i < tasks.Length; i++)
-                    {
-                        tasks[i] = cws.ReceiveAsync(recvSegment, cts.Token);
-                    }
+                    tasks[i] = cws.ReceiveAsync(recvSegment, cts.Token);
+                }
 
-                    Task.WaitAll(tasks);
+                Exception e = Record.Exception(() => Task.WaitAll(tasks));
+
+                if (e == null)
+                {
                     Assert.Equal(WebSocketState.Open, cws.State);
                 }
-                catch (AggregateException ag)
+                else
                 {
-                    foreach (var ex in ag.InnerExceptions)
+                    AggregateException ae = Assert.IsType<AggregateException>(e);
+
+                    Assert.All(ae.InnerExceptions, ex =>
                     {
+                        Assert.True(ex is InvalidOperationException || ex is WebSocketException,
```

Comment:
I prefer doing this, because it directly states the condition we're actually testing (that the exception is one of those types), rather than having it be inferred from the leftover conditions.  Something equivalent to an `Assert.AnyOf(T actual, IEnumerable<T> expected)`, although that's probably wrong from a testing point of view ("should know exactly what the result is").

Ideally, tests should potentially be rewritten to separate the situations under which the two exceptions are thrown... however, the code being tested involves threading/cancellation, and changing it this way would involves additional work.